fuel-vm/src/storage/interpreter.rs

```diff
@@ -145,17 +146,23 @@ pub trait InterpreterStorage:
         &mut self,
         contract: &ContractId,
         key: &Bytes32,
-        value: &Bytes32,
-    ) -> Result<Option<Bytes32>, Self::DataError> {
-        StorageMutate::<ContractsState>::insert(self, &(contract, key).into(), value)
+        value: &StorageData,
+    ) -> Result<Option<StorageData>, Self::DataError> {
+        // StorageWrite::<ContractsState>::write(self, &(contract, key).into(),
+        // value.into())
```

Comment:
Maybe these comments should be removed?

Reply:
This part specifically is unclear in my mind. We want to use a higher level of abstraction  for the `ContractsState` table, on top of `StorageMutate` and `StorageInspect` by implementing `StorageWrite` and `StorageRead`. This should consolidate the read and write patterns for vector based storage. I'm not sure I understand how/why that's used in this specific context of Merkle storage.

Reply:
Summarizing from the call earlier today - it sounds like we need to implement StorageRead and StorageWrite for the contracts state table. This will also allow us to structure the table using `Vec<u8>` or `[u8]` under the hood and avoid the need for the newtype wrapper.

Reply:
We will still use the new wrapper type to provide some convenience functions for converting between types, as well as to control the canonical serialization implementation while we need a fixed size.